tapir/shifts/views/views.py

```diff
@@ -309,3 +314,210 @@ def get(self, request, *args, **kwargs):
         call_command("run_freeze_checks")
         messages.info(request, _("Frozen statuses updated."))
         return super().get(request, args, kwargs)
+
+
+class SolidarityShiftUsed(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        date = datetime.now()
+        tapir_user = get_object_or_404(TapirUser, pk=kwargs["pk"])
+        solidarity_shift = SolidarityShift.objects.filter(is_used_up=False)[0]
```

Comment:
This would cause an HTTP 500 error if the filter returns an empty QuerySet, I think a 400 BAD REQUEST would be more appropriate. I'm not sure but I think you can just `return HttpResponseBadRequest("...")`

Reply:
I think it shouldn't be possible for this to happen, since the Button for using solidarity shifts is only enabled when there is a solidarity shift available? But I'll implement it anyways. Surely doesn't hurt to check (:

Reply:
The database may change in between the page getting rendered and the button being clicked, and overall it's a bit closer to "never trust user input" to check on the backend.